docs-src/how-to-restructure-sort.md

```diff
@@ -14,10 +14,73 @@ kernelspec:
 How to sort arrays and inner arrays
 ===================================
 
-**This is a stub:** I intend to write this article, but haven't yet.
+Sorting puts Awkward Array elements in an ordered sequence: numeric or lexicographical, ascending or descending.
 
-If you need it soon, create an issue saying so and I'll make it a higher priority.
+By default, `ak.sort` sorts the elements in non-descending order. The order of equivalent elements is guaranteed to be preserved.
 
-[![](img/github-issues-documentation.png)](https://github.com/scikit-hep/awkward-1.0/issues/new?assignees=&labels=docs&template=documentation.md&title=)
+The back-end is an `std::stable_sort` with complexity `O(N·log(N)2)`, where `N = std::distance(first, last)` applications of a comparator. If an additional memory is available, then the complexity is `O(N·log(N))`.
 
-The text of your issue doesn't have to be much more than a link to this page, so I can be sure which page you're referring to. If you add details about how and why you need it, however, I may be able to tailor the text to help you more.
+```{code-cell} ipython3
+import awkward as ak
+ak.sort(array = ak.Array([[2, 5, 3], [], [None, 0], [1]]))
+```
+
+Sorting in axis
+---------------
+
+An optional `axis` parameter defines an axis along which to sort. The default is `axis=-1`, which is the last axis.
+
+Awkward Array axes are defined similar to Numpy axes. Unlike other operations, sorting does not support `axis=None`.
+
+[![](img/sorting-axis.svg)]
+
+In Akward Array, `axis=0` is the "first" axis and `axis=-1` is the last axis.
+
+```{code-cell} ipython3
+import awkward as ak
+ak.sort(array = ak.Array([[2, 5, 3], [], [None, 0], [1]]), axis=0)
+```
+
+and
+
+```{code-cell} ipython3
+import awkward as ak
+ak.sort(array = ak.Array([[2, 5, 3], [], [None, 0], [1]]), axis=1)
+```
+An empty range is not sorted.
+
+The results make a lot of sense when you understand how the Awkward Array axis work.
+
+Sorting `None` values
+---------------------
+
+Sorting places `None` values at the end of an ordered sequence.
+
+```{code-cell} ipython3
+import awkward as ak
+array = ak.Array([[None, None, 3, 2], [1]])
+ak.sort(array)
+```
+
+`None` values are dropped when sorting a one dimentinal option type array:
+
+```{code-cell} ipython3
+import awkward as ak
+array = ak.Array([None, None, 1, -1, 30])
+ak.sort(array)
```

Comment:
@jpivarski - I wonder if dropping the `None`s is a bug:
```python
>>> import awkward as ak
>>> array = ak.Array([None, None, 1, -1, 30])
>>> array
<Array [None, None, 1, -1, 30] type='5 * ?int64'>
>>> ak.sort(array)
<Array [-1, 1, 30] type='3 * int64'>
```
because they are preserved in a higher dimensional array:
```python
>>> array = ak.Array([[None, None, 3, 2], [1]])
>>> ak.sort(array)
<Array [[2, 3, None, None], [1]] type='2 * var * ?int64'>

```

Reply:
NaN is not the same thing as None (which Pandas is finding, and that's why they're trying to deprecate all the methods that looked for NaN in favor of option-types). However, in

```python
>>> narr =np.array([np.nan, np.nan, 1, -1, 30])
>>> np.sort(narr)
array([-1.,  1., 30., nan, nan])
```

it looks like it's being sorted the same way. If NaN is insensitive to the choice of `ascending=True` vs `ascending=False`, that's to be expected because all comparisons with NaN result in false:

```python
>>> a = np.nan
>>> b = np.nan
```

would return false from `a < b` and also from `b < a`. So if ascending puts Nones at one end of the list and descending puts Nones at the other, but the same isn't true for NaN, I'd say that's okay. NaN is not None.

For the other case,

```python
>>> import awkward as ak
>>> array = ak.Array([None, None, 1, -1, 30])
>>> array
<Array [None, None, 1, -1, 30] type='5 * ?int64'>
>>> ak.sort(array)
<Array [-1, 1, 30] type='3 * int64'>
```

yes, that is a bug. It's probably a case of the implementation assuming that you're not at top-level, but in this case, you're at top-level. Perhaps it could be resolved by adding a case to check for top-level and do the right thing, or it could follow some of the other methods in wrapping the input array with a length-1 RegularArray, adjusting the `axis` (positive or negative), applying the operation, and then unwrapping the outer RegularArray when done. Wrapping/unwrapping a RegularArray is an _O(1)_ operation, and then the inner logic only ever has to deal with the "I'm nested inside of another node" case.